File: Data Structures & Algorithms/same-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool dfs(TreeNode* p, TreeNode* q,bool &ans)
    {
        if(ans==false)
        {
            return false;
        }
        if(!p && !q)
        {
            ans = true;
            return true;
        }
        if((p == nullptr && q != nullptr) || (p != nullptr && q == nullptr) || (p->val != q->val))
        {
            ans = false;
            return false;
        }
        else
        {
            dfs(p->left,q->left,ans);
            dfs(p->right,q->right,ans);
        }

        
        

        return ans;
    }
    bool isSameTree(TreeNode* p, TreeNode* q) {
        bool ans = true;
        dfs(p,q,ans);
        return ans;
        
        
    }
};

File: Data Structures & Algorithms/binary-tree-from-preorder-and-inorder-traversal/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func buildTree(preorder []int, inorder []int) *TreeNode {
    if len(preorder) == 0{
        return nil
    }
    rootVal := preorder[0]
    i := 0
    for ind,val := range inorder{
        if val == rootVal{
            i = ind
            break
        }
    }
    leftSub := buildTree(preorder[1:i+1], inorder[:i])
    rightSub := buildTree(preorder[i+1:], inorder[i+1:])
    return &TreeNode{Val:rootVal, Left:leftSub, Right:rightSub}
}